src/components/ComputeModal.js

```diff
@@ -1006,19 +1009,19 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
       div({ style: { ...gridStyle, gridTemplateColumns: '0.25fr 8.5rem 1fr 5.5rem 1fr 5rem', marginTop: '1.5rem' } }, [
         h(IdContainer, [
           id => div({ style: { gridColumnEnd: 'span 3' } }, [
-            label({ htmlFor: id, style: computeStyles.label }, ['Location']),
+            label({ htmlFor: id, style: computeStyles.label }, ['Location ']),
+            versionTag('Beta', { color: colors.primary(1.5), backgroundColor: 'white', border: `1px solid ${colors.primary(1.5)}` }),
             h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
-              'Cloud environments run in the same region as the workspace bucket and cannot be changed at this time.'
+              'Cloud environments run in the same region as the workspace bucket and can be changed as a beta feature.'
             ]),
             div({ style: { marginTop: '0.5rem' } }, [
               h(Select, {
                 id,
-                isDisabled: true, // Initially releasing regionality supporting only us-central1 region
-                // isDisabled: computeExists, // Can't update location of existing environments
+                isDisabled: computeExists || isUSLocation(bucketLocation) || sparkMode === 'cluster', // Can't update location of existing environments
```

Comment:
1. Why do we need `isUSLocation(bucketLocation)`?
2. Should we add `sparkMode === 'master'` to exclude Dataproc clusters with a single node?

Reply:
1) For workspaces with buckets in us-central1 or us multi-regional, we don't want to enable a location choice. `isUSLocation` is a helper function returns true if the workspace bucket is us-central1 or us multi-regional.
2) My understanding is that single node dataproc clusters still work. The bug with dataproc clusters is caused by misaligned IP ranges preventing the VMs in the clusters from communicating properly.